myst_nb/parser.py

```diff
@@ -37,6 +41,25 @@ def parse(self, inputstring, document):
         # de-serialize the notebook
         ntbk = nbf.reads(inputstring, nbf.NO_CONVERT)
 
+        self.reporter = document.reporter
+        self.config = self.default_config.copy()
+
+        try:
+            source_path = document.settings.env.doc2path(document.settings.env.docname)
+            dest_path = document.settings.env.app.outdir
+            new_cfg = document.settings.env.config.myst_config
+            self.config.update(new_cfg)
+        except AttributeError:
+            pass
+            
+        ## add outputs to notebook from the cache
+        if hasattr(document.settings.env, 'path_cache'):
+            path_cache = document.settings.env.path_cache
+            ntbk = add_notebook_outputs(source_path, ntbk, path_cache, dest_path)
+        else:
+            # If we explicitly did not wish to cache, then just execute the notebook
+            ntbk = execute(ntbk)
```

Comment:
I think there should be a config option to not do any execution. This would also be useful for testing, because then we could keep the original tests, like complex_ouputs, that are more focussed on the output creation, and add new notebooks/tests specific to execution

Reply:
you are right, I was planning to create it as well. Will create one with default False.